net/src/client/tcp/actors/jsonrpc.rs

```diff
@@ -394,3 +465,26 @@ fn subscription_topic_from_request(request: &Request) -> String {
         .expect("Subscriptions should always have a topic")
         .expect("Subscription topics should always be String")
 }
+
+/// Pick a random element from a list, avoiding "twice-in-a-row" repetition when possible.
+fn pick_random<T>(input: &[T], old: T) -> Option<T>
+where
+    T: Clone + cmp::PartialEq,
+{
+    match input.len() {
+        0 => None,
+        1 => Some(input[0].clone()),
+        _ => {
+            let mut pick;
+            loop {
+                pick = input.choose(&mut rand::thread_rng())?.clone();
+
+                if pick != old {
+                    break;
+                }
+            }
```

Comment:
You can avoid the reroll like this:

* Choose 1 out of N-1 instead of 1 out of N, for example making it impossible to get the element at position 0
* If the chosen item is equal to the expected item, return the element at position 0 instead
* Otherwise, return the random item

This assumes that there is only 1 element with the same value as `old`.

```suggestion
            let pick = input[1..].choose(&mut rand::thread_rng())?.clone();

            if pick == old {
                return Some(input[0].clone());
            }
```